### ISSUE TITLE
JDBC driver doesn't work with Kuberneteres Pinot cluster

### ISSUE THREAD

**Issue by KKcorps** [apache/incubator-pinot#7166]
The JDBC driver is not able to run queries if pinot is deployed in kubernetes cluster.
The driver uses controller URL in the connection and then fetches the broker list from controller for the tenant specified in JDBC properties 
Now what's happening is that in clusters such as k8s, the controller returns the broker hostname and not IP/ingress url 
```bash
curl -X GET "http://localhost:9000/v2/brokers/tenants/DefaultTenant?state=ONLINE" -H "accept: application/json"
```
Queries cannot be executed against hostname since it is unresolvable to the drivers which are running outside the Kubernetes cluster.

**Comment by KKcorps** [comment]
@Jackie-Jiang @kishoreg

**Comment by xiangfu0** [comment]
You are right, pinot client expects to use pinot controller for getting table management info and broker to query.
Here is one workaround solution: modify pinot client to take a flag/param to be controller-only mode, so it can query pinot directly on the controller. Then JDBC-driver also need to pass that flag to the pinot-client.

**Comment by xiangfu0** [comment]
another workaround is to write a proxy service to take this if we don't want a lot of traffics to the controller. I know Uber is doing this.

**Comment by KKcorps** [comment]
@xiangfu0 The proxy deployment solution is scalable but less pragmatic for users who will be using the driver. Either we should provide an out of the box proxy which can be easily deployed OR expose query endpoint in controller which may not be scalable.
